system-addon/lib/ActivityStreamMessageChannel.jsm

```diff
@@ -125,6 +127,36 @@ this.ActivityStreamMessageChannel = class ActivityStreamMessageChannel {
     return null;
   }
 
+  /**
+   * sendToPreloaded - Sends an action to the preloaded browser, if any
+   *
+   * @param  {obj} action A redux action
+   */
+  sendToPreloaded(action) {
+    const preloadedBrowser = this.getPreloadedBrowser();
+    if (preloadedBrowser && action.data) {
+      try {
+        preloadedBrowser.sendAsyncMessage(this.outgoingMessageName, action);
+      } catch (e) {
+        // The preloaded page has already been consumed, so just ignore.
+      }
+    }
+  }
+
+  /**
+   * getPreloadedBrowser - Retrieve the port of a preloaded browser, if it exists
+   *
+   * @return {string|null} The unique id of the target, if it exists.
+   */
+  getPreloadedBrowser() {
+    for (let port of this.channel.messagePorts) {
+      if (port.browser.getAttribute("preloadedState") === "preloaded") {
+        return port;
```

Comment:
Ah so it might just find the first preloaded browser from the first window it checks and then not update the other ones? An array of preloaded browser ports and a message to each would fix this issue?